metabase/metabase
# Fix `type/PK`  -> `type/Name` and `type/FK` field remapping in the QB

Fixes https://github.com/metabase/metabase/issues/45063
Fixes https://github.com/metabase/metabase/issues/7231
Fixes https://github.com/metabase/metabase/issues/37955

BE changes:
- Hydrates `name_field` for query metadata.

MBQL lib API changes:
- Fixes `dimensions` parsing. This makes `external`, i.e. `type/FK` remapping work out of the box. 
- Makes `type/PK` -> `type/Name` field remapping work.

FE changes:
- Makes `SearchValuePicker` handle mixed remapping options. Adds loading state.
- **Note that selected values are displayed as raw values and this is out of scope for this PR.**

How to verify type/PK -> type/Name remapping:
- Make sure that in Table Metadata for `People.Name` `Entity Name` semantic type is set
- Set `Filtering on this field` for `People.ID` to `A list of all values`
- Open `People` table. Click on `ID` -> Filter on this column. You should see a list of remapped values.
- Set `Filtering on this field` for `People.ID` to `Search box`
- Open `People` table. Click on `ID` -> Filter on this column.  Start typing some name. You should get autocomplete. Please note that the placeholder `Search by ID` is misleading and this is something that we need to fix separately.

How to verify type/FK -> column remapping:
- Open Table Metadata -> `Orders.Product ID` -> Click `Use original value` -> `Use foreign key` -> Select `Products.Title`.
- Set `Filtering on this field` to `A list of all values`
- Open `Orders` table. Click on `Product ID` -> Filter on this column. You should see a list of product names.
- Do the same with `Search box`.
